presto-spi/src/main/java/io/prestosql/spi/connector/AggregateFunction.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import io.prestosql.spi.expression.ConnectorExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+public class AggregateFunction
+{
+    private final String functionName;
+    private final Type outputType;
+    private final List<ConnectorExpression> inputs;
+    private final List<Map.Entry<String, SortOrder>> sortOrder;
+    private final boolean isDistinct;
+    private final Optional<String> filter;
```

Comment:
Not sure about this, Why would this need to be a variable given the filter on the agg side is just a Symbol?

Reply:
Symbola (engine things) are mapped to Variables in ConnectorExpression.
Using Variable here would give semantics to this field -- connector would know it needs to check assignments to know which column handle this is.

Reply:
BTW why isn't filter a `Optional<ConnectorExpression>` (or `ConnectorExpression` without Optional,  using `new Constant(true, BOOLEAN)` instead of empty).
@Parth-Brahmbhatt  @martint did you discuss this bit already?

Reply:
We haven't discussed this, originally I did not have mask or filter as I couldn't quite understand how those were used. I will figure out how to get the corresponding connector expression from the symbol and update the PR today.

Reply:
we haven't discussed this, I originally had no filters or masks as I did not completely understand what masks are for. @martint asked to add the filter in this PR in the first pass so the connectors have to explicitly decide to not handle them. For the current rule given it will only be matched when agg is followed by scan node the only possible place to look for filter symbol is tablescan assignment so I think it makes sense to just map it to Variable for now. I can't think of a scenario where we match this rule and it has a filter but the symbol is not in tablescan assignment so I have not added a check for that If you think we should add a defensive check I can add a condition. 

I personally prefer Optional<ConnectorExpression> so I have kept it as is, let me know if you would like me to remove Optional.